scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/job/factories/spi/model/validator/ModelValidator.java

```diff
@@ -86,48 +89,33 @@ public String validate(String parameterValue, ModelValidatorContext context) thr
-            } else if (uppercaseModel.startsWith(ShortParserValidator.SHORT_TYPE)) {
-                return new ShortParserValidator(removePrefix(model));
-            } else if (uppercaseModel.startsWith(BooleanParserValidator.BOOLEAN_TYPE)) {
-                return new BooleanParserValidator(removePrefix(model));
-            } else if (uppercaseModel.startsWith(URLParserValidator.URL_TYPE)) {
-                return new URLParserValidator(removePrefix(model));
-            } else if (uppercaseModel.startsWith(URIParserValidator.URI_TYPE)) {
-                return new URIParserValidator(removePrefix(model));
-            } else if (uppercaseModel.startsWith(DateTimeParserValidator.DATETIME_TYPE)) {
-                return new DateTimeParserValidator(removePrefix(model));
-            } else if (uppercaseModel.startsWith(ListParserValidator.LIST_TYPE)) {
-                return new ListParserValidator(removePrefix(model));
-            } else if (uppercaseModel.startsWith(RegexpParserValidator.REGEXP_TYPE)) {
-                return new RegexpParserValidator(removePrefix(model));
-            } else if (uppercaseModel.startsWith(ModelFromURLParserValidator.MODEL_FROM_URL_TYPE)) {
-                return new ModelFromURLParserValidator(removePrefix(model));
-            } else if (uppercaseModel.startsWith(CRONParserValidator.CRON_TYPE)) {
-                return new CRONParserValidator(removePrefix(model));
-            } else if (uppercaseModel.startsWith(SPELParserValidator.SPEL_TYPE)) {
-                return new SPELParserValidator(removePrefix(model));
-            } else if (uppercaseModel.startsWith(CatalogObjectParserValidator.CATALOG_OBJECT_TYPE)) {
-                return new CatalogObjectParserValidator(removePrefix(model));
-            } else if (uppercaseModel.startsWith(JSONParserValidator.JSON_TYPE)) {
-                return new JSONParserValidator(removePrefix(model));
-            } else if (uppercaseModel.startsWith(NotEmptyParserValidator.NOT_EMPTY_TYPE)) {
-                return new NotEmptyParserValidator(removePrefix(model));
-            } else {
-                throw new ModelSyntaxException("Unrecognized type in model '" + model + "'");
-            }
-        } else {
+        if (!uppercaseModel.startsWith(PREFIX)) {
             return null;
         }
+        uppercaseModel = removePrefix(uppercaseModel);
+        for (ModelType type : ModelType.values()) {
+            if (uppercaseModel.startsWith(type.name())) {
+                try {
+                    return (ParserValidator) type.getTypeParserValidator()
+                                                 .getDeclaredConstructor(String.class)
+                                                 .newInstance(removePrefix(model));
+                } catch (InstantiationException | IllegalAccessException | InvocationTargetException
+                        | NoSuchMethodException e) {
+                    if (e instanceof InvocationTargetException) {
```

Comment:
You could catch InvocationTargetException without the other Exceptions it would avoid to check the instantiation of the exception. However by doing so you avoid duplicate the final throw. Was it your purpose there ?

Reply:
Yes, that's the reason.